### ISSUE TITLE
The hardblock was added to prevent Explorer from crashing when ExplorerPatcher is installed.

### ISSUE THREAD

**Issue by Michellehdr** [valinet/ExplorerPatcher#3053]
Hi Team,

The hardblock was added to prevent Explorer from crashing when ExplorerPatcher is installed. You only need to adjust the setup exe name to get around the block in your new version. We will continue to block ANY version that crashes Explorer.

Please let me know if you have any question.

Thanks
Michelle

**Comment by symphony** [comment]
Is this why my taskbar is transparent now with win10 style?
![image](https://github.com/valinet/ExplorerPatcher/assets/24639525/a542e7c5-a875-43fb-950d-dab70e6a408e)

**Comment by Michellehdr** [comment]
@Speedy37 could you please take a look at this bug? Thanks!

**Comment by Michellehdr** [comment]
@pyrates999, this issue only repro on GE build. Could you please check if you are using GE? @Amrsatrio FYI

**Comment by Amrsatrio** [comment]
This issue is not repro on latest 26090 build. I have my changes to make the 11 taskbar the default on all Windows 11 versions unpushed.

**Comment by Michellehdr** [comment]
@Amrsatrio updated the description. We removed Windows taskbar code from Windows GE build and above. If user use ExplorerPatcher on those Windows version, crash will occur.

**Comment by Michellehdr** [comment]
@Amrsatrio please kindly confirm if you received my comment above. Please let me know if you have any question. Thanks a lot!

**Comment by Michellehdr** [comment]
@pyrates999, @Amrsatrio please kindly confirm if you received my comment above. Please let me know if you have any question. Thanks a lot!

**Comment by pyrates999** [comment]
Hi @Michellehdr as I'm not one of the developers here, I'll let @Amrsatrio reply to this :)

**Comment by dxgldotorg** [comment]
As for code removal, is @Michellehdr digging in their heels about not allowing moving or resizing the Taskbar?

**Comment by Amrsatrio** [comment]
Hi @Michellehdr, I'm here to let you know that I've received your comment. Sorry for taking a while, I've been having other errands lately. I will make a new release today or tomorrow to make the Windows 10 taskbar no longer an option when run on 26002+.

**Comment by Michellehdr** [comment]
@Amrsatrio thanks for the update. Looking forward to the new release. And long termly, it will be great if you could update the app to stop using undocumented code to prevent current and future issues.

**Comment by dxgldotorg** [comment]
Since the legacy Explorer Taskbar has been killed off, please consider feedback that re-enables more Windows 10 Taskbar features, especially resizing and moving the Taskbar.

Most people have widescreen displays, and some have ultrawide displays, and thus vertical space is a bit more at a premium, hence why some people like to put their Taskbar on the side of the display.

**Comment by Michellehdr** [comment]
Hi Amr,

Is it possible to have a private channel for us to have a communication of the bug? I saw a lot of people replying me and someone ask who am I. I’m also wondering who is the dedicated developer for ExplorerPatcher.

Thanks!
Haifang

**Comment by Amrsatrio** [comment]
Hi, sorry for taking so long. I have pushed the code where the 10 taskbar is no longer available on 26002 to comply with your decisions. It would be very great if versions 22621.3527.65.1 and above can be whitelisted.

Feel free to close this issue if the 65.1 no longer causes crashes on your unreleased GE builds.

**Comment by Bec-de-Xorbin** [comment]
> Hi, sorry for taking so long. I have pushed the code where the 10 taskbar is no longer available on 26002 to comply with your decisions. It would be very great if versions 22621.3527.65.1 and above can be whitelisted.
> 
> Feel free to close this issue if the 65.1 no longer causes crashes on unreleased GE builds.

If you read the opening post carefully, they said that you'll need to rename the ep_setup.exe to something else to lift the block.

**Comment by Bec-de-Xorbin** [comment]
> Hi, sorry for taking so long. I have pushed the code where the 10 taskbar is no longer available on 26002 to comply with your decisions. It would be very great if versions 22621.3527.65.1 and above can be whitelisted.
> 
> Feel free to close this issue if the 65.1 no longer causes crashes on unreleased GE builds.

@Amrsatrio If you read the opening post carefully, they said that you'll just need to rename the ep_setup.exe to something else to lift the block. As far as I can see the new release is still ep_setup.exe. Also it seems that they have retracted the claim that Win10 taskbar code is removed, at least it doesn't appear anymore on the edited post... So as long as it doesn't crash it can inculde the Windows 10 taskbar code.

**Comment by dxgldotorg** [comment]
> > Hi, sorry for taking so long. I have pushed the code where the 10 taskbar is no longer available on 26002 to comply with your decisions. It would be very great if versions 22621.3527.65.1 and above can be whitelisted.
> > Feel free to close this issue if the 65.1 no longer causes crashes on unreleased GE builds.
> 
> @Amrsatrio If you read the opening post carefully, they said that you'll just need to rename the ep_setup.exe to something else to lift the block. As far as I can see the new release is still ep_setup.exe. Also it seems that they have retracted the claim that Win10 taskbar code is removed, at least it doesn't appear anymore on the edited post... So as long as it doesn't crash it can inculde the Windows 10 taskbar code.

That would explain why the new version blew up when I tried and that I had to rename the uninstaller to get rid of it.

**Comment by Amrsatrio** [comment]
@Bec-de-Xorbin I will consider this later. 

> Also it seems that they have retracted the claim that Win10 taskbar code is removed, at least it doesn't appear anymore on the edited post

I have confirmations that we're being blocked due to EP fighting the DisableWin10Taskbar feature flag. Now the Windows 10 taskbar is no longer here, so is the code that fights DisableWin10Taskbar.

**Comment by VaiCaDep0893** [comment]
we can simply just [replace windows 11's explorer with windows 10's to have the old taskbar](https://www.youtube.com/watch?v=52YV657Rd1Q) again

**Comment by Amrsatrio** [comment]
@VaiCaDep0893 Stability wise I disagree. Doesn't work if you have more than one input languages.

**Comment by dxgldotorg** [comment]
> @VaiCaDep0893 Stability wise I disagree. Doesn't work if you have more than one input languages.

Not to mention that Microsoft would probably frown on this project contining Microsoft-owned files.

**Comment by David-Castro** [comment]
i do have a backup of windows 11 explorer 22h2
use that while you can
[explorer.zip](https://github.com/valinet/ExplorerPatcher/files/15172491/explorer.zip)

**Comment by pyrates999** [comment]
Where would this explorer.exe go?

**Comment by markran** [comment]
If this means there won't be any way to have to get the taskbar on Win11 on the side as well as all the other functionality MSFT has chosen to remove (Quick Launch, etc), then Windows 11 will no longer be usable at our company.

Due to the uncertainty around MSFT's intentions to fulfill their earlier promises about implementing full taskbar functionality in Win11 we've avoided upgrading our computers. Unfortunately, our IT vendor is only offering laptops with Win11 recently. Prior to deployment we'll need to downgrade all new laptops to a Win10 Long Term Maintenance version or the last Win11 version that can be made usable with ExplorerPatcher (and then freeze them on that version). 

Does anyone know the final working version of Windows 11?

**Comment by David-Castro** [comment]
> If this means there won't be any way to have to get the taskbar on Win11 on the side as well as all the other functionality MSFT has chosen to remove (Quick Launch, etc), then Windows 11 will no longer be usable at our company.
> 
> Due to the uncertainty around MSFT's intentions to fulfill their earlier promises about implementing full taskbar functionality in Win11 we've avoided upgrading our computers. Unfortunately, our IT vendor is only offering laptops with Win11 recently. Prior to deployment we'll need to downgrade all new laptops to a Win10 Long Term Maintenance version or the last Win11 version that can be made usable with ExplorerPatcher (and then freeze them on that version).
> 
> Does anyone know the final working version of Windows 11?

Probably 23H2, because i haven't had any issues running it.

**Comment by pyrates999** [comment]
Still recommend against it.

**Comment by David-Castro** [comment]
> Still recommend against it.

it's the only way

**Comment by pyrates999** [comment]
That's not secure to accept executables from strangers.

**Comment by dxgldotorg** [comment]
> > Still recommend against it.
> 
> it's the only way

Still, EP needs to put out a build that is fully 26100 compliant to be exempted from the block, and the option of injecting older builds of copyrighted MS binaries is probably not all that legal.

**Comment by vlascik** [comment]
How about adding a function to backup the system's current explorer.exe. After a new version without Taskbar10 gets shipped, just ask to overwrite it from a backup. License-wise, it's a clean approach, and keeps the functionality.

Also, fsck what MSFT likes, some of their recent decisions (e.g. killing off WSA after getting it almost perfect, but way before it had a real chance to catch on) are just mind-bogglingly stupid.

**Comment by Amrsatrio** [comment]
@vlascik Nuh uh. It's not as modular as you thought.

**Comment by vlascik** [comment]
> @vlascik Nuh uh. It's not as modular as you thought.

It seems to be. Saw this video yet? https://www.youtube.com/watch?v=52YV657Rd1Q

**Comment by morrow95** [comment]
Reading over this and seems like the Win 10 taskbar is gone from here on out - is that correct?  My main reason for using it all this time was the ability to NOT group windows which is something that is standard with Win 11.  Not sure if anyone else deals with this, but for me personally and the work I do, it is a huge pita when taskbar icons are grouped together.  Have to hover over, figure out which window from the popups I actually want given the small thumbnail previews, and so on - not productive at all for me at least.

If the Win 10 taskbar is removed is there any possibility of an 'ungrouping' option for the Win 11 taskbar?

**Comment by dxgldotorg** [comment]
> Reading over this and seems like the Win 10 taskbar is gone from here on out - is that correct? My main reason for using it all this time was the ability to NOT group windows which is something that is standard with Win 11. Not sure if anyone else deals with this, but for me personally and the work I do, it is a huge pita when taskbar icons are grouped together. Have to hover over, figure out which window from the popups I actually want given the small thumbnail previews, and so on - not productive at all for me at least.
> 
> If the Win 10 taskbar is removed is there any possibility of an 'ungrouping' option for the Win 11 taskbar?

The ungrouping options were added back in either 23H2 or 22H2.

**Comment by Azarien** [comment]
Classic taskbar is the main reason I use ExplorerPatcher. If Win10 taskbar is gone for good, I will have to find an alternative, be it going back to Windows 10 until there's a good solution to the taskbar problem.

What I specifically rely on is a taskbar with small icons, labels, no grouping, and resizeable to mutiple rows (I use 2 or 3 rows).

**Comment by jerl0** [comment]
> Classic taskbar is the main reason I use ExplorerPatcher. If Win10 taskbar is gone for good, I will have to find an alternative, be it going back to Windows 10 until there's a good solution to the taskbar problem.
> 
> What I specifically rely on is a taskbar with small icons, labels, no grouping, and resizeable to mutiple rows (I use 2 or 3 rows).

Same here.

**Comment by pyrates999** [comment]
> > @vlascik Nuh uh. It's not as modular as you thought.
> 
> It seems to be. Saw this video yet? https://www.youtube.com/watch?v=52YV657Rd1Q

That is for an advanced user and to me looks like it will cause your windows OS to break the moment a windows update is released that replaces explorer.exe.  So I don't think that will ever be implemented in EP because of the amount of bugs that it will cause with future windows updates.

**Comment by vlascik** [comment]
> That is for an advanced user and to me looks like it will cause your windows OS to break the moment a windows update is released that replaces explorer.exe. So I don't think that will ever be implemented in EP because of the amount of bugs that it will cause with future windows updates.

Copying a file is not an "advanced operation" for "advanced user". I would also congratulate you on having a crystal ball and seeing a future so clearly, but everything ExplorerPatcher does breaks with Windows updates already (that is why they patch insider versions). I would even go on saying that replacing one exe file with an older version (an original file made and signed by Microsoft too) is way less invasive than what EP does at the moment.

**Comment by pyrates999** [comment]
The source code for EP is available.  So feel free to try to add this feature.

**Comment by morrow95** [comment]
> The ungrouping options were added back in either 23H2 or 22H2.

@dxgldotorg - I had no idea they added this feature in.  It had been soo long since release I stopped bothering to check.

**Comment by vlascik** [comment]
@Amrsatrio the goal could be to use Windows 11 explorer (the last version that works with EP) on Windows 11. Yes, the APIs might drift over time, but it also might work for years just fine. Which might be just enough for MSFT to fix their taskbar.

I've checked the video and haven't seen anything wrong with the input panel, but even if it there was, who cares, it's a cosmetic problem and there's always alt+shift.

All in all, Windows 10 explorer.exe got its last update 2 years ago (last feature work was probably done 3-4 years ago too), and it still works well enough (outside Japan anyway). The situation will be better with Windows 11 explorer.

So. If the choices are having nothing or a mild discomfort of slightly outdated explorer.exe, I'd chose the mild discomfort. This could be a way to stretch this project for couple more years further. 

But, as always, as maintainers you're free to kill this project at this point too. ¯\_(ツ)_/¯.

**Comment by Amrsatrio** [comment]
@vlascik Let's talk about this further in private through our Discord.

https://discord.gg/gsPcfqHTD2

**Comment by hondacbr600** [comment]
> Classic taskbar is the main reason I use ExplorerPatcher. If Win10 taskbar is gone for good, I will have to find an alternative, be it going back to Windows 10 until there's a good solution to the taskbar problem.
> 
> What I specifically rely on is a taskbar with small icons, labels, no grouping, and resizeable to mutiple rows (I use 2 or 3 rows).

Same.

**Comment by dxgldotorg** [comment]
> > The ungrouping options were added back in either 23H2 or 22H2.
> 
> @dxgldotorg - I had no idea they added this feature in. It had been soo long since release I stopped bothering to check.

I personally enabled it the instant it appeared in Insider Preview.  Also set multiple monitor taskbars to show the buttons on the taskbar where window is open.

**Comment by Maxstate69** [comment]
Can someone explain to me what's going on, in very basic terms? I am interested in learning but the conversation is a bit too high level.

**Comment by Amrsatrio** [comment]
@Maxstate69 

1. Microsoft removed the system tray from the old taskbar in 25921, suggesting that they're killing the old taskbar in the latest Windows 11 builds.
2. Microsoft added the DisableWin10Taskbar feature flag in 26002 to further deter 3rd party customization programs from unearthing them. The feature flag (which is enabled by default and can be disabled using ViVeTool) when enabled causes explorer.exe to crash if the code related to the old taskbar is reached. I usually call it "crash bombs".
3. ExplorerPatcher and StartAllBack introduced code that disarms the bombs, and Microsoft doesn't seem to like it maybe because the disarming code does not work in unreleased builds internal to Microsoft or maybe because Microsoft doesn't want programs to do so in general.
4. ExplorerPatcher removed the code to disarm the bombs and removed the Windows 10 taskbar as an option, followed by StartAllBack doing the same roughly a week later.

**Comment by Maxstate69** [comment]
> @Maxstate69
> 
> 1. Microsoft removed the system tray from the old taskbar in 25921, suggesting that they're killing the old taskbar in the latest Windows 11 builds.
> 2. Microsoft added the DisableWin10Taskbar feature flag in 26002 to further deter 3rd party customization programs from unearthing them. The feature flag (which is enabled by default and can be disabled using ViVeTool) when enabled causes explorer.exe to crash if the code related to the old taskbar is reached. I usually call it "crash bombs".
> 3. ExplorerPatcher and StartAllBack introduced code that disarms the bombs, and Microsoft doesn't seem to like it maybe because the disarming code does not work in unreleased builds internal to Microsoft or maybe because Microsoft doesn't want programs to do so in general.
> 4. ExplorerPatcher removed the code to disarm the bombs and removed the Windows 10 taskbar as an option, followed by StartAllBack doing the same roughly a week later.
> 
> TLDR: It's the time for the 10 taskbar to say goodbye from Windows 11, says Microsoft.

Thank you. That sucks. Can you speculate as to the reason Ms is doing this?

**Comment by Amrsatrio** [comment]
If you have been reading the news around Windows lately, you will see that Microsoft has been in a nuking spree of legacy components and remnants in Windows 11 24H2.

**Comment by Maxstate69** [comment]
Hmm, seems reasonable... Too bad. It would be nice if they left people the *choice* to use what they want, for instance by putting it back if they want...

**Comment by pyrates999** [comment]
> Hmm, seems reasonable... Too bad. It would be nice if they left people the _choice_ to use what they want, for instance by putting it back if they want...

Unless it's an option in windows itself, I don't think MS intended anyone to use the windows 10 taskbar in windows 11.

MS also wants to show ads in the start menu, etc and EP bypassed that.

**Comment by Maxstate69** [comment]
I understand that and the whole discussion around it, but really, it's my computer and I wish I could do what I wanted with it, whether ms intends as much or not.

I'm a privacy lawyer and am trying to contribute to EU legislators coming down on this sort of behavior. Windows is the de facto standard OS in billions of cases. We should have more control over something that our day to day infrastructure relies upon. We're lucky to have had things like explorerpatcher in the meantime.

In the worst of all possible worlds, I'd love for Ms to just tell me how much they want me to pay for a debloated, af-free, telemetry-free and customizable version of windows. I'll give them what they want and the transaction should be over at that point. But this sort of drug dealer behavior is really not my cup of tea.

**Comment by Amrsatrio** [comment]
For telemetry, I am sure that it really helps during the development phase of features. But I do not agree if that cannot be completely disabled. The rest, I completely agree with you. I am stuck to Windows due to Fortnite and Genshin not being available on other desktop OSes like MacOS and Linux, but Microsoft's decisions lately has been pushing me to rethink my career on these games.

**Comment by Amrsatrio** [comment]
@Maxstate69 I have more facts that you might find interesting, please drop me a DM through EP's Discord server: https://discord.gg/gsPcfqHTD2

**Comment by Maxstate69** [comment]
> @Maxstate69 I have more facts that you might find interesting, please drop me a DM through EP's Discord server: https://discord.gg/gsPcfqHTD2

Thank you, I will. It will take some time, as I am in the middle of an international move...

**Comment by dxgldotorg** [comment]
> I understand that and the whole discussion around it, but really, it's my computer and I wish I could do what I wanted with it, whether ms intends as much or not.
> 
> I'm a privacy lawyer and am trying to contribute to EU legislators coming down on this sort of behavior. Windows is the de facto standard OS in billions of cases. We should have more control over something that our day to day infrastructure relies upon. We're lucky to have had things like explorerpatcher in the meantime.
> 
> In the worst of all possible worlds, I'd love for Ms to just tell me how much they want me to pay for a debloated, af-free, telemetry-free and customizable version of windows. I'll give them what they want and the transaction should be over at that point. But this sort of drug dealer behavior is really not my cup of tea.

If you are in fact a privacy lawyer, have you looked at the Diagnostic Data Viewer?  This can reveal all the telemetry data sent to Microsoft.  You might if concerned about telemetry want to look at the kind of data Bambu Lab 3D printers collect as it has tried to hide such info until reverse engineered.

That said, customizability of built-in UI elements may be outside the scope of your law practice.

**Comment by hondacbr600** [comment]
> > Hmm, seems reasonable... Too bad. It would be nice if they left people the _choice_ to use what they want, for instance by putting it back if they want...
> 
> Unless it's an option in windows itself, I don't think MS intended anyone to use the windows 10 taskbar in windows 11.
> 
> MS also wants to show ads in the start menu, etc and EP bypassed that.

Which is stupid, I'd rather move the favorites/quick launch folder to desktop and use the shortcuts from there, than use the start menu. 
MS Power Toys has really nice search function, so I will deny stupid start menu from showing up completely.

**Comment by YellowAfterlife** [comment]
Wanted to stop by to say... _gosh darn_

I could see this as the final step as the Win10 taskbar is made all but redundant, but it's been a few years and Win11 taskbar is still quite lacking if you'd rather see the labels on a bunch of windows at once (be it through moving it to the side of the screen or adding a second row when necessary)

Replacing important system components with third-party solutions is certainly fun, but doesn't feel like that should be necessary... especially for features that were part of the OS for a couple decades before that.

**Comment by SIBtomcat** [comment]
Am I right to understand that the taskbar in the new version will only be large? "Taskbar icons size" is "medium" (large - actually). And there is no way to make the taskbar small?

**Comment by hondacbr600** [comment]
> small

Yes. And the QuickLaunch toolbars also

**Comment by pyrates999** [comment]
I don't see the point of the quick launch toolbar anymore since you can now pin applications on the taskbar itself.

**Comment by hondacbr600** [comment]
> I don't see the point of the quick launch toolbar anymore since you can now pin applications on the taskbar itself.

![image](https://github.com/valinet/ExplorerPatcher/assets/15723130/2d597d34-12be-4ea2-b836-49d112f08b4d)

**Comment by detcordtech** [comment]
For me, one of the main reasons to be using EP was to put my taskbar on top or side whenever I want with win10 taskbar/battery/calendar/etc.

Microsoft seems to have jumped on the brainless bubbly bandwagon - 
1. Changing themes, adding rounded corners -- no one wanted this
2. Adding too many zones (old zones were enough)
3. Liking their own features so much that they announce a new version of windows (11) going back on their own word when they said "10 will be the last one"
4. Removing the most well-known features and saying it's "too difficult" to maintain them down the road from their own OS
5. Getting annoyed when awesome devs (you all) manage to make their (closed source) code work with all features
6. Being pathetic and petty, adding crash bombs in their own code, to prevent people from bringing old and well made features back, since this proves, "it's actually not that difficult"

It makes me laugh, when big tech companies spend actual development time into adding "crash bombs" in their own old code just so that the tiny 3% of users using these tweaks won't be able to get what they want. Why are they so despicably petty...? 😂 

If you can, please fight their intentional garbage and reconsider adding win10 taskbar, even if it's under a separate branch/repo/name/etc and even if it's a bit unstable. I grew up with windows (xp, 7, 10, 11) and it's sad to see it in such a state.

EP (with w10 taskbar) is the only one that can make Win11 daily drivable for me... please save it if possible 🙏

Until then I'm staying on the last pre-release before this change was made (22621.3296.64.3), running W11 23H2 22631.3296 hoping this feature will be revived/saved someday. Win updates are blocked so it should remain stable for now.

**Comment by Azarien** [comment]
> Changing themes, adding rounded corners -- no one wanted this

If they don't change the looks people would complain lack of any innovation, because nothing's visibly changed.

**Comment by Maxstate69** [comment]
You can innovate without losing features, so I believe this is a false
dichotomy

tor. 6. juni 2024, 12:58 skrev Azarien ***@***.***>:

**Comment by A7exSchin** [comment]
Just stumbled over this thread while looking for a way to bring the task bar to the left side because **it just makes sense** on widescreen displays. It reminds me of the time when Microsoft killed the [EdgeDeflector](https://github.com/da2x/EdgeDeflector) and forced users to use Bing and Edge...

If these changes go live it is just a bigger argument for using Linux for work and something like AtlasOS for Gaming.

**Comment by doterik** [comment]
I guess this is the right thread for my current problem.

### _Explorer always restarts. An information window shows that EP will be shut down due to crashes, but nothing happens, Explorer keeps restarting._

Windows 10.0.22635.3785 (latest symbols file downloaded)
EP 22621.3527.65.2 (automatically installed some time ago)

I have restarted the computer a few times but with the same result.

When I tried to uninstall via the command `ep_setup.exe /uninstall`, a window showed that the uninstall was completed. In connection with this, I received several virus warnings and files quarantined which, for me, are difficult to manage without a 'Desktop'. (The only thing that happened was the window showing that EP is to be disabled now is gone.)

On new reboots, Explorer continues to restart repeatedly. By killing the process, I can run a command prompt, but if I was content with that I might as well install Linux.

**What can I do?**
**How do I get Windows to work again?**

**Comment by pyrates999** [comment]
Update to the latest pre-release: https://github.com/valinet/ExplorerPatcher/releases/tag/22621.3527.65.5_6f6666b

**Comment by doterik** [comment]
@pyrates999 Thank you for your response.

Unfortunately, the version provided did not resolve the issue.

The behavior remains unchanged; Explorer is caught in an endless restart loop, accompanied by a recurring notification that EP will be disabled - though nothing changes.

I attempted to uninstall EP once more, yet Explorer persists in restarting. This time, the EP notification window is absent; instead, there’s a continuous alert about the symbols file has been downloaded, yet the taskbar remains invisible. I was compelled to terminate the process again.

Following a reboot, the same issue recurs. The notification has disappeared, but Explorer continues to restart endlessly.

Could EP have replaced any Windows files, or might there be registry changes that can be undone?

**Comment by Amrsatrio** [comment]
@doterik I have not tested that particular Beta build yet, but will check it out soon to see if something is fixable. In the meantime please delete C:\Windows\dxgi.dll.

**Comment by doterik** [comment]
@Amrsatrio That was a quick reply...

Don't have it in \Windows, I suppose you mean \Windows\System32.

What about \Windows\SysWOW64 ?

**Comment by Amrsatrio** [comment]
Hm, weird, if Explorer keeps crashing even without that file present it means there is something wrong with that build or your Windows install. Also please do not delete C:\Windows\System32\dxgi.dll, it is a crucial system file.

**Comment by doterik** [comment]
@Amrsatrio Hmm...mysterious...

Should I copy the one from _\Windows\System32_ to _\Windows_ ?

Could running a `sfc /scannow` help? (I suppose it at least can't hurt.)

**Comment by Amrsatrio** [comment]
No, do not do that as well. I don't know honestly, worth a try. I've had similar Explorer crash loops even without EP on beta builds in the past. I can say that Beta builds are unstable sometimes.

**Comment by doterik** [comment]
@Amrsatrio Ok, thanks for your answer.

The latest beta version (.3785) started normally. I think it was in connection with when the symbols file was downloaded, at a later time, and then a restart of Explorer that the problems arose. I don't think there was anything wrong with the symbols file, but something may have gone wrong with the restart of Explorer.

I'll try running sfc and we'll see...

**Comment by doterik** [comment]
@Amrsatrio SFC didn't help, unfortunately, not a single error.

Come to think of it, isn't _\Windows\dxgi.dll_ from EP, and if so, it was probably removed during uninstall (quite in order).

Thanks anyway for your support.

**Comment by raxpyraxp** [comment]
So basically as I understand - there will be no way to stick the taskbar to the top of the screen?
I see that Win11 taskbar has no such option now.
Do someone knows an alternative for this? I don't mind switching to Win11 taskbar if that's how things are going now. Guess I can get used to larger icons, but bottom taskbar is a little bit painful change for me

**Comment by robpomeroy** [comment]
> Do someone knows an alternative for this?

I don't mean this flippantly: this has made me seriously consider finally switching to a Linux desktop. I would just need to do an audit to check which software I use would need to run in a Win11 VM.

I've had the taskbars at the sides of my dual monitors for so many years that the bottom taskbar feels like a very retrograde step.

**Comment by raxpyraxp** [comment]
> this has made me seriously consider finally switching to a Linux desktop

I legit want give it a try again. I only moved from Linux as a daily driver because of the online games that I've been playing (don't want to risk my accounts to be banned for cheating as some of them don't play nice with wine/proton) and due to some instability on my hardware that might've been fixed by now

**Comment by TheKidRock** [comment]
@Michellehdr Only reason I use EP is to display 2 clocks on the Taskbar. Default clock allows 2 clocks only on hovering but I used Atomic Alarm Clock which lets me display 2 clocks to display 2 different timezones on Taskbar Tray. 
On default Windows 11 taskbar, I get this error "Trayclockwclass not found" but after installing EP and enabling Windows 10 taskbar, atomic clock used to work fine. Now, in Windows 11 24H2, neither the clock is working nor EP's Windows 10 taskbar. 
So, is it possible to let me have 2 clocks display on Windows 11 taskbar? Is there a solution for Trayclockwclass error? I haven't found any other clocks that have features to display 2 clocks on taskbar.

**Comment by Veyilla** [comment]
Is there a way to minimize the taskbar and its icons? I find the fixed size unsightly and annoying. That's actually why I downloaded the program, I was never interested in the Win10 taskbar. I would just like it to look smaller again and only show me the time, without the date, as it used to be possible to set under win10.


My operating system is as follows: 

Edition: Windows 11 Pro
Version: 24H2
Operating system build: 26120.994
Performance: Windows Feature Experience Pack 1000.26100.6.0

**Comment by pyrates999** [comment]
> Is there a way to minimize the taskbar and its icons? I find the fixed size unsightly and annoying. That's actually why I downloaded the program, I was never interested in the Win10 taskbar. I would just like it to look smaller again and only show me the time, without the date, as it used to be possible to set under win10.
> 
> My operating system is as follows:
> 
> Edition: Windows 11 Pro Version: 24H2 Operating system build: 26120.994 Performance: Windows Feature Experience Pack 1000.26100.6.0

Try this: https://www.geeksforgeeks.org/how-to-change-taskbar-size-in-windows-11/

**Comment by mrplow** [comment]
two rows of applications is not possible with 24h2 as far as I can see. Reg tweaks only resize the bar/icons but still don't allow proper resizing. <sarcastic>Good job MS!</sarcastic>

**Comment by AJolly** [comment]
> I don't see the point of the quick launch toolbar anymore since you can now pin applications on the taskbar itself.

Quick launch gives you hotkeys - Win+1, Win+2, Win+3 etc

**Comment by Veyilla** [comment]
> > Is there a way to minimize the taskbar and its icons?
> 
> Try this: https://www.geeksforgeeks.org/how-to-change-taskbar-size-in-windows-11/

This is not more functional in the new Versions of Windows 11...

**Comment by levitation** [comment]
@Michellehdr

> @Amrsatrio thanks for the update. Looking forward to the new release. And long termly, it will be great if you could update the app to stop using undocumented code to prevent current and future issues.

With all due respect, I am wondering whether you are sure that your request to not use undocumented code complies with EU DMA (https://digital-markets-act.ec.europa.eu/index_en)? Essentially, Microsoft as a operating system developer should not create functions which provide unfair advantage to Microsoft's other software. The functionalities Microsoft implements in the operating system should be available for all software developers, not only to Microsoft. Since these undocumented functionalities are inevitably part of the operating system, then it is fair that other developers use them just as freely as Microsoft themselves would use them.

**Comment by pyrates999** [comment]
I think undocumented code is code that can change with every new update that is released for windows.  It's not guaranteed to stay the same.  So the compatibility can change if that code changes.

**Comment by daltroaugusto** [comment]
> @Amrsatrio thanks for the update. Looking forward to the new release. And long termly, it will be great if you could update the app to stop using undocumented code to prevent current and future issues.

Maybe if you LISTEN quickly to your users and stop capping Windows features to align to MICROSOFT values and opinions on UI, everything would be great. Another good shot would be to DOCUMENT YOUR entire codebase; anyway, this ir OUR software that WE got to OUR own use.

f#ck M5

**Comment by kkiris** [comment]
I get this error when I try to update ExplorerPatcher… Is this the block or something different? 
![image](https://github.com/user-attachments/assets/5ca7205f-f8d5-4df1-b35a-7d68d2cd6fff)

If it matters, I'm running Windows 11 Pro 23H2 (build 22635.3936), and my current ExplorerPatcher version is 22621.3810.66.1.

**Comment by pyrates999** [comment]
> I get this error when I try to update ExplorerPatcher… Is this the block or something different? ![image](https://private-user-images.githubusercontent.com/76605732/351603207-5ca7205f-f8d5-4df1-b35a-7d68d2cd6fff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MjgxNjMsIm5iZiI6MTcyMTgyNzg2MywicGF0aCI6Ii83NjYwNTczMi8zNTE2MDMyMDctNWNhNzIwNWYtZjhkNS00ZGYxLWIzNWEtN2Q2OGQyY2Q2ZmZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI0VDEzMzEwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiOWFiYmFhMzgwZDlhZmIyZjBmNWE4MjRjNWE4MzE4NDI3NDEwNjY4YjM4NmRiNWQ1MTI3NGFhMmVmMjExYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xAVRexJuyFwBe33yp6hEzZDdpdxja6K0dTLdko9wZak)
> 
> If it matters, I'm running Windows 11 Pro 23H2 (build 22635.3936), and my current ExplorerPatcher version is 22621.3810.66.1.

yes, it is windows defender blocking the pre-release of EP.
